src/v/raft/replicate_entries_stm.cc

```diff
@@ -30,13 +30,13 @@ using namespace std::chrono_literals;
 ss::future<append_entries_request> replicate_entries_stm::share_request() {
     // one extra copy is needed for retries
     return with_semaphore(_share_sem, 1, [this] {
-        return details::foreign_share_n(std::move(_req.batches), 2)
+        return details::foreign_share_n(std::move(_req->batches), 2)
```

Comment:
does this patch need to be sequenced before the patch that converts `_req` to a pointer type?

Reply:
yes, I will change this.